src/runtime_src/core/include/ert.h

```diff
@@ -547,41 +609,45 @@ struct cu_cmd_state_timestamps {
 /**
  * Opcode types for commands
  *
- * @ERT_START_CU:       start a workgroup on a CU
- * @ERT_START_KERNEL:   currently aliased to ERT_START_CU
- * @ERT_CONFIGURE:      configure command scheduler
- * @ERT_EXEC_WRITE:     execute a specified CU after writing
- * @ERT_CU_STAT:        get stats about CU execution
- * @ERT_START_COPYBO:   start KDMA CU or P2P, may be converted to ERT_START_CU
- *                      before cmd reach to scheduler, short-term hack
- * @ERT_SK_CONFIG:      configure soft kernel
- * @ERT_SK_START:       start a soft kernel
- * @ERT_SK_UNCONFIG:    unconfigure a soft kernel
- * @ERT_START_KEY_VAL:  same as ERT_START_CU but with key-value pair flavor
- * @ERT_START_DPU:      instruction buffer command format
- * @ERT_CMD_CHAIN:      command chain
+ * @ERT_START_CU:          start a workgroup on a CU
+ * @ERT_START_KERNEL:      currently aliased to ERT_START_CU
+ * @ERT_CONFIGURE:         configure command scheduler
+ * @ERT_EXEC_WRITE:        execute a specified CU after writing
+ * @ERT_CU_STAT:           get stats about CU execution
+ * @ERT_START_COPYBO:      start KDMA CU or P2P, may be converted to ERT_START_CU
+ *                         before cmd reach to scheduler, short-term hack
+ * @ERT_SK_CONFIG:         configure soft kernel
+ * @ERT_SK_START:          start a soft kernel
+ * @ERT_SK_UNCONFIG:       unconfigure a soft kernel
+ * @ERT_START_KEY_VAL:     same as ERT_START_CU but with key-value pair flavor
+ * @ERT_START_DPU:         instruction buffer command format
+ * @ERT_CMD_CHAIN:         command chain
+ * @ERT_START_IPU:         instruction buffer command format on IPU format
+ * @ERT_START_IPU_PREEMPT: instruction buffer command with preemption format on IPU
  */
 enum ert_cmd_opcode {
-  ERT_START_CU      = 0,
-  ERT_START_KERNEL  = 0,
-  ERT_CONFIGURE     = 2,
-  ERT_EXIT          = 3,
-  ERT_ABORT         = 4,
-  ERT_EXEC_WRITE    = 5,
-  ERT_CU_STAT       = 6,
-  ERT_START_COPYBO  = 7,
-  ERT_SK_CONFIG     = 8,
-  ERT_SK_START      = 9,
-  ERT_SK_UNCONFIG   = 10,
-  ERT_INIT_CU       = 11,
-  ERT_START_FA      = 12,
-  ERT_CLK_CALIB     = 13,
-  ERT_MB_VALIDATE   = 14,
-  ERT_START_KEY_VAL = 15,
-  ERT_ACCESS_TEST_C = 16,
-  ERT_ACCESS_TEST   = 17,
-  ERT_START_DPU     = 18,
-  ERT_CMD_CHAIN     = 19,
+  ERT_START_CU          = 0,
+  ERT_START_KERNEL      = 0,
+  ERT_CONFIGURE         = 2,
+  ERT_EXIT              = 3,
+  ERT_ABORT             = 4,
+  ERT_EXEC_WRITE        = 5,
+  ERT_CU_STAT           = 6,
+  ERT_START_COPYBO      = 7,
+  ERT_SK_CONFIG         = 8,
+  ERT_SK_START          = 9,
+  ERT_SK_UNCONFIG       = 10,
+  ERT_INIT_CU           = 11,
+  ERT_START_FA          = 12,
+  ERT_CLK_CALIB         = 13,
+  ERT_MB_VALIDATE       = 14,
+  ERT_START_KEY_VAL     = 15,
+  ERT_ACCESS_TEST_C     = 16,
+  ERT_ACCESS_TEST       = 17,
+  ERT_START_DPU         = 18,
+  ERT_CMD_CHAIN         = 19,
+  ERT_START_IPU         = 20,
```

Comment:
Is "ERT_START_IPU" just for aie2p?
Adding some comment so we know the difference between "ERT_START_IPU" and "ERT_START_DPU" would be nice.

Reply:
I wish function here https://github.com/Xilinx/XRT/blob/1779b1f7660ef2f93e4e24d943a64bbaf61f754b/src/runtime_src/core/common/api/xrt_module.cpp#L769
is simple and clear enough.